### ISSUE TITLE
Warning on `check-param-names` on Vue emit (after bump to 62.6)

### ISSUE THREAD

**Issue by FelixNumworks** [gajus/eslint-plugin-jsdoc#1663]
## Expected behavior

When I was on v62.5.0, this didn't raise any warning:
_(this in an emit definition in the script part of my Vue3 component)_

```ts
const emit = defineEmits<{
  /**
   * Fired when an editable field is submitted.
   * @param index the index of the editable field
   */
  submitField: [index: number];
}>();
```

## Actual behavior

Since I bumped to 62.6.0, I have the following warning on the line `@param index ....`

```
@param "index" does not match an existing function parameter. eslint(jsdoc/check-param-name)
```

## ESLint Config

```js
      'jsdoc/require-jsdoc': [
        'warn',
        {
          minLineCount: 10,
          require: {
            ArrowFunctionExpression: false,
            ClassDeclaration: true,
            ClassExpression: false,
            FunctionDeclaration: true,
            FunctionExpression: false,
            MethodDefinition: true,
          },
        },
      ],
      'jsdoc/check-alignment': 'warn',
      'jsdoc/check-param-names': 'warn',
      'jsdoc/check-tag-names': 'warn',
      'jsdoc/implements-on-classes': 'warn',
      'jsdoc/no-types': 'warn',
      'jsdoc/require-description': 'warn',
      'jsdoc/require-param': 'warn',
      'jsdoc/require-param-description': 'warn',
      'jsdoc/require-param-name': 'warn',
      'jsdoc/require-returns': 'warn',
      'jsdoc/require-returns-check': 'warn',
      'jsdoc/require-returns-description': 'warn',
```

## Environment

- Node version: 24.13.0
- ESLint version: 9.39.2
- `eslint-plugin-jsdoc` version: 62.6.0
